runtime/src/pcr.rs

```diff
@@ -0,0 +1,110 @@
+// Licensed under the Apache-2.0 license
+
+use crate::Drivers;
+use caliptra_common::mailbox_api::{
+    IncrementPcrResetCounterReq, MailboxResp, MailboxRespHeader, QuotePcrsReq, QuotePcrsResp,
+};
+use caliptra_drivers::{
+    cprintln, hand_off::DataStore, CaliptraError, CaliptraResult, Ecc384PrivKeyIn, KeyReadArgs,
+    PcrBank, PcrId,
+};
+use core::ops::{Index, IndexMut};
+use enum_map::{EnumArray, EnumMap};
+use zerocopy::{transmute, FromBytes};
+
+pub struct PcrResetCounter {
+    counter: enum_map::EnumMap<PcrId, u32>,
```

Comment:
This seems like it could just be a `[u32; ALL_PCR_IDS.len()]`. And then populating the quote response is trivial.

Reply:
A previous version did that, but @sree-revoori1, wrote the following in #1111 (which is based on this PR):
"I think you might need a guard here to prevent potential panic since usize::from(id) could be out of array bounds" when accessing elements, so I looked into an enum array. I can revert it back.

Reply:
You can run cargo test test_panic_missing within the runtime directory to check if your change induces potential panic. Then for the guard, it can just be an if statement checking that usize::from(id) is within the bounds of the array, and if it isn't it returns some sort of error.

Reply:
No panics created. The array of counters always has the exact length of the number of PcrId and PcrId is now used as an index, so no panic is ever possible.

Reply:
Sounds good, thanks for checking.